core/config/toml/types.go

```diff
@@ -1434,14 +1439,71 @@ func (r *ExternalRegistry) setFrom(f *ExternalRegistry) {
 	}
 }
 
+type GatewayConnectorConfig struct {
+	NodeAddress               *string
+	DonId                     *string
+	Gateways                  []ConnectorGatewayConfig
+	WsClientConfig            *gatewayConfig.WebSocketClientConfig
+	AuthMinChallengeLen       *int
+	AuthTimestampToleranceSec *uint32
+}
+
+func (r *GatewayConnectorConfig) setFrom(f *GatewayConnectorConfig) {
+	if f.NodeAddress != nil {
+		r.NodeAddress = f.NodeAddress
+	}
+
+	if f.DonId != nil {
+		r.DonId = f.DonId
+	}
+
+	// TODO: verify this copy by reference is ok, or does array need to be copied by value
+	if f.Gateways != nil {
```

Comment:
It's fine to just copy by reference here, but when returning the array (i.e. in the interface implementation) you probably do want to copy it